dev/cd/after-tag-push

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+if [ -z "${GIT_REF}" ]; then
+  echo "GIT_REF must be set"
+  exit 1
+fi
+
+if [[ ! "${GIT_REF}" =~ ^refs/tags/.* ]]; then
+  echo "GIT_REF=${GIT_REF} is not of the expected format refs/tags/*"
+  exit 1
+fi
+
+TAG=${GIT_REF/refs\/tags\//}
+echo "TAG is ${TAG}"
+
+VERSION=${TAG##*/}
+PREFIX=${TAG%/*}
+echo "PREFIX is ${PREFIX}, VERSION is ${VERSION}"
+
+if [[ ! -d "${PREFIX}" ]]; then
+  echo "Directory ${PREFIX} is not found"
+  exit 1
+fi
+
+export IMAGE_TAG=${VERSION}
+cd ${PREFIX}
+make push
```

Comment:
It looks like this assumes each function has it's own Makefile with a `push` target defined. I don't think all of the functions meet that assumption - is this intentional?

I also think we could simplify the boilerplate by using a generic make target that builds any function with a provided name. That way each directory doesn't have to implement it's own make target. Maybe for flexibility we could check if the directory has a `Makefile` and use that if it exists, but default to the generic one if it doesn't exist?

Reply:
Agreed an opinionated builder sounds like the way to go for these functions. I expect in 99% of use cases we want to do pretty much the same thing, which is build a simple go binary and put it in an image (and possibly add docs, etc).

My understanding is that we essentially already have an [opinionated builder](https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/744) with the current make targets and scripts. Is there a gap in the current implementation that we are trying to solve?